profile/src/v100/entity/persona/persona_data/entry_kinds/name.rs

```diff
@@ -0,0 +1,209 @@
+use crate::prelude::*;
+
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    PartialEq,
+    Hash,
+    Eq,
+    derive_more::Display,
+    derive_more::Debug,
+    uniffi::Record,
+)]
+#[display("{}", self.full())]
+#[debug("{}", self.full())]
+#[serde(rename_all = "camelCase")]
+pub struct Name {
+    pub variant: Variant,
+    pub family_name: String,
+    pub given_names: String,
+    pub nickname: String,
+}
+
+impl Name {
+    pub fn new(
+        variant: Variant,
+        family_name: impl AsRef<str>,
+        given_names: impl AsRef<str>,
+        nickname: impl AsRef<str>,
+    ) -> Result<Self> {
+        let family_name = family_name.as_ref().trim().to_string();
+        let given_names = given_names.as_ref().trim().to_string();
+        let nickname = nickname.as_ref().trim().to_string();
+        if family_name.is_empty()
+            || given_names.is_empty()
+            || nickname.is_empty()
+        {
+            return Err(CommonError::InvalidDisplayNameEmpty);
```

Comment:
wrong error type, create own, also three different, for granularity, for family name, given names and nickname.